nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -531,6 +532,8 @@ FAIL:
 			if res.createEval || len(res.allowReplacements) != 0 {
 				w.createBatchedUpdate(res.allowReplacements, allocIndex)
 			}
+
+			allocsCh = w.getAllocsCh(allocIndex)
```

Comment:
The position within the block is subtle so it's probably worth leaving a comment indicating why it is where it is:

```suggestion
			// only start a new blocking query if we haven't returned early
			allocsCh = w.getAllocsCh(allocIndex)
```